src/Test/L1/Mock/FakeAgentPluginManager.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using Agent.Sdk;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+
+namespace Microsoft.VisualStudio.Services.Agent.Tests.L1.Worker
+{
+    public class FakeAgentPluginManager : AgentPluginManager
+    {
+        public override void Initialize(IHostContext hostContext)
+        {
+            _taskPlugins.Remove("Agent.Plugins.Repository.CheckoutTask, Agent.Plugins");
+            _taskPlugins.Add("Microsoft.VisualStudio.Services.Agent.Tests.L1.Worker.FakeCheckoutTask, Test");
```

Comment:
This is where you inject any plugin mocks. Right now we're just mocking checkout, but its easily extensible to other plugins (e.g. artifacts, etc...)

Reply:
Maybe a comment to that effect would be good in this file.